### PR TITLE
[Fix] Fix type hint in file_client

### PR DESCRIPTION
**Pull request by imabackstabber**
## Motivation

To fix some type hint in file_client 

## Modification

1. Add mypy in precommit.yaml to check type hint
2. Add type hint in  `file_client.py`,also add some type hint in related file imported by `fle_client.py`

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/tensorrt/__init__.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def is_tensorrt_available():
         return False
 
 
-__all__ = []
+__all__ = []  # type: ignore
```
Comment (zhouzaida) [inline mmcv/tensorrt/__init__.py:16]: What errors will be printed if not adding `#type: ignore`?

Comment (imabackstabber) [reply mmcv/tensorrt/__init__.py:16]: it says "error: Type of __all__ must be "Sequence[str]", not "List[List[str]]""

Comment (imabackstabber) [reply mmcv/tensorrt/__init__.py:16]: > What errors will be printed if not adding `#type: ignore`?

it says "error: Type of __all__ must be "Sequence[str]", not "List[List[str]]""

Comment (zhouzaida) [reply mmcv/tensorrt/__init__.py:16]: Please update the PR by merging mmcv master branch

**Review by zhouzaida — APPROVED** [review]

**Review by teamwong111 — APPROVED** [review]

**Pull request merged**